examples/rust/raw_mesh/src/main.rs

```diff
@@ -34,21 +34,33 @@ impl From<GltfPrimitive> for Mesh3D {
             vertex_positions,
             vertex_colors,
             vertex_normals,
-            vertex_texcoords: _, // TODO(cmc) support mesh texturing
+            vertex_texcoords: _, // TODO(cmc): support mesh texturing
         } = primitive;
 
-        let raw = RawMesh3D {
-            albedo_factor: albedo_factor.map(Vec4D).map(Into::into),
-            indices: indices.map(|i| i.into()),
-            vertex_positions: vertex_positions.into_iter().flatten().collect(),
-            vertex_normals: vertex_normals.map(|normals| normals.into_iter().flatten().collect()),
-            vertex_colors: vertex_colors
-                .map(|colors| colors.into_iter().map(|c| c.to_u32()).collect()),
-        };
+        let mut mesh = Mesh3D::new(vertex_positions);
+
+        if let Some(indices) = indices {
+            assert!(indices.len() % 3 == 0);
+            let triangle_indices = indices.chunks_exact(3).map(|xyz| (xyz[0], xyz[1], xyz[2]));
```

Comment:
This is again confusing. `xyz` is so strongly associated with the three spatial dimensions.

```suggestion
            let triangle_indices = indices.chunks_exact(3).map(|tri| (tri[0], tri[1], tri[2]));
```